### DISCUSSION TITLE
Flickr images not showing

### DISCUSSION THREAD

**Discussion by grunta56** [mcguirepr89/BirdNET-Pi#489] [Q&A] [answered]
Anyone else lost their Flickr feed in the last 2 weeks or so? I don't think its a Flickr issue as Flickr dashboard indicates that API calls are being made as I would expect. The images are just not showing up in BirdNet Pi.

**Comment by issakandji** [comment]
Loading Images from Flickr here and have no issue.
Could you provide logs from birdnetPi maybe we'll check together what's going wrong.
Have you checked if traffic is not blocked at you side, network issue ?

**Reply by grunta56** [reply]
Thankyou. I have checked local network and no issues that I can detect. I'm not sure which log files you will need?

**Comment by ehpersonal38** [comment, marked as answer]
Can confirm things seem to be working as expected on my installation and on Patrick's public installation.

https://farm66.static.flickr.com/65535/51028450922_6e5c22a557.jpg

Does this image load when you click it?

**Reply by grunta56** [reply]
Yes that loads thankyou

**Reply by ehpersonal38** [reply]
Okay let's make sure your API key is working. Try accessing this URL in your browser, but replace the XXXXXXXXXXXXXXXXXXXXXXX with your Flickr API key.
https://www.flickr.com/services/rest/?method=flickr.photos.search&api_key=XXXXXXXXXXXXXXXXXXXXXXX&text=northern%20cardinal&per_page=5&media=photos&format=json&nojsoncallback=1

You should get a response like this if all is working correctly:
```
{"photos":{"page":1,"pages":23553,"perpage":5,"total":117765,"photo":[{"id":"52281929068","owner":"183506946@N03","secret":"d89cb0e55c","server":"65535","farm":66,"title":"Northern Cardinal","ispublic":1,"isfriend":0,"isfamily":0},{"id":"52281864331","owner":"96366044@N02","secret":"bf9fbffc79","server":"65535","farm":66,"title":"Northern Cardinal","ispublic":1,"isfriend":0,"isfamily":0},{"id":"52281768731","owner":"49563725@N06","secret":"02fe85270d","server":"65535","farm":66,"title":"Disagreement","ispublic":1,"isfriend":0,"isfamily":0},{"id":"52281403416","owner":"18739614@N00","secret":"543291ab7c","server":"65535","farm":66,"title":"Northern Cardinal","ispublic":1,"isfriend":0,"isfamily":0},{"id":"52281241141","owner":"128466905@N04","secret":"900651c4b6","server":"65535","farm":66,"title":"MFP_7414 Northern Cardinal","ispublic":1,"isfriend":0,"isfamily":0}]},"stat":"ok"}
```

**Reply by grunta56** [reply]
Yes, that works with a response like that

**Reply by ehpersonal38** [reply]
Hmm, okay. Could you make the Overview page show more verbose error logs with this command: `sed -i 's/E_ERROR/E_ALL/' ~/BirdNET-Pi/scripts/overview.php`

and then try on another browser to see if images show up then, or if any weird messages show up? Also make sure your API key in your config has no spaces in the beginning or front.

**Reply by grunta56** [reply]
Thankyou. There are no spaces in API key text. Upon running sed command the following warnings and notices appear on the Overview page. I wonder given that it was prior to this working well for a couple of months would it be best at this stage to just do a complete fresh install? I don't want to lose the data collected to date - 

Warning: ini_set(): A session is active. You cannot change the session module's ini settings at this time in /home/pi/BirdNET-Pi/scripts/overview.php on line 4
Warning: session_set_cookie_params(): Cannot change session cookie parameters when session is active in /home/pi/BirdNET-Pi/scripts/overview.php on line 5
Notice: session_start(): A session had already been started - ignoring in /home/pi/BirdNET-Pi/scripts/overview.php on line 6
Notice: Undefined index: only_name in /home/pi/BirdNET-Pi/scripts/overview.php on line 57
Notice: Undefined index: images in /home/pi/BirdNET-Pi/scripts/overview.php on line 78
Warning: array_column() expects parameter 1 to be array, null given in /home/pi/BirdNET-Pi/scripts/overview.php on line 78
Warning: array_search() expects parameter 2 to be array, null given in /home/pi/BirdNET-Pi/scripts/overview.php on line 78
Notice: Undefined index: images in /home/pi/BirdNET-Pi/scripts/overview.php on line 80
Notice: Trying to access array offset on value of type null in /home/pi/BirdNET-Pi/scripts/overview.php on line 80

**Reply by ehpersonal38** [reply]
Okay, I compared the messages you got with the ones I get on a Flickr-working installation. These are the ones unique to your system:
```
Notice: Undefined index: images in /home/pi/BirdNET-Pi/scripts/overview.php on line 78
Warning: array_column() expects parameter 1 to be array, null given in /home/pi/BirdNET-Pi/scripts/overview.php on line 78
Warning: array_search() expects parameter 2 to be array, null given in /home/pi/BirdNET-Pi/scripts/overview.php on line 78
Notice: Undefined index: images in /home/pi/BirdNET-Pi/scripts/overview.php on line 80
Notice: Trying to access array offset on value of type null in /home/pi/BirdNET-Pi/scripts/overview.php on line 80
```

It looks like `$_SESSION['images']` is null, which is strange because a few lines above in the code (41-43) I explicitly set that variable... I think this is most likely due to some weird session issue. I will look into this more tomorrow. Can you confirm that accessing your BirdNET-Pi in an incognito window (that will clear all sessions and give us a clean slate) also produces this result?

**Reply by ehpersonal38** [reply]
Actually just had a thought - are cookies enabled on your browser? Also try disabling any ad blockers.

**Reply by grunta56** [reply]
OK. No ad blockers and cookies cleared. I confirm that in a private window the messages as above do still appear.

**Reply by ehpersonal38** [reply]
Okay I'm pretty stumped on this one... I just made a few changes to the Overview.php which should tell us more, if you switch to the debug branch: `bash ~/BirdNET-Pi/scripts/update_birdnet.sh -b flickr-debug`

Then if you'll copy the notices and warning messages again I will hopefully have a better idea of what's gone wrong.

**Reply by grunta56** [reply]
OK thankyou. Here we are -
RAN THIS -
mybirdnetpi:~/BirdNET-Pi$ bash ~/BirdNET-Pi/scripts/update_birdnet.sh -b flickr-debug
+ sudo -u pi git -C /home/pi/BirdNET-Pi reset --hard
HEAD is now at b5dabf9 Update play.php
+ set +x
+ sudo -u pi git -C /home/pi/BirdNET-Pi fetch origin flickr-debug
From https://github.com/mcguirepr89/BirdNET-Pi
 * branch            flickr-debug -> FETCH_HEAD
+ set +x
+ sudo -u pi git -C /home/pi/BirdNET-Pi switch -C flickr-debug --track origin/flickr-debug
Branch 'flickr-debug' set up to track remote branch 'flickr-debug' from 'origin'.
Switched to a new branch 'flickr-debug'
+ set +x
^Cmybirdnetpi:~/BirdNET-Pi$ ^


GOT FROM OVERVIEW PAGE (removed flickr address, replaced with xxxxxxxxx) - 

Warning: ini_set(): A session is active. You cannot change the session module's ini settings at this time in /home/pi/BirdNET-Pi/scripts/overview.php on line 4

Warning: session_set_cookie_params(): Cannot change session cookie parameters when session is active in /home/pi/BirdNET-Pi/scripts/overview.php on line 5

Notice: session_start(): A session had already been started - ignoring in /home/pi/BirdNET-Pi/scripts/overview.php on line 6


array(3) { ["behind"]=> string(4) "with" ["FLICKR_FILTER_EMAIL"]=> string(13) "XXXXXXXXX@N08" ["images"]=> array(4) { [0]=> array(5) { [0]=> string(17) "Eastern_Spinebill" [1]=> string(37) "https://farm.static.flickr.com//_.jpg" [2]=> NULL [3]=> string(27) "https://flickr.com/photos//" [4]=> string(26) "https://flickr.com/people/" } [1]=> array(5) { [0]=> string(17) "Common_Bronzewing" [1]=> string(37) "https://farm.static.flickr.com//_.jpg" [2]=> NULL [3]=> string(27) "https://flickr.com/photos//" [4]=> string(26) "https://flickr.com/people/" } [2]=> array(5) { [0]=> string(12) "Little_Raven" [1]=> string(37) "https://farm.static.flickr.com//_.jpg" [2]=> NULL [3]=> string(27) "https://flickr.com/photos//" [4]=> string(26) "https://flickr.com/people/" } [3]=> array(5) { [0]=> string(5) "Galah" [1]=> string(37) "https://farm.static.flickr.com//_.jpg" [2]=> NULL [3]=> string(27) "https://flickr.com/photos//" [4]=> string(26) "https://flickr.com/people/" } } } 1: SESSION is NOT null
array(3) { ["behind"]=> string(4) "with" ["FLICKR_FILTER_EMAIL"]=> string(13) "XXXXXXXXX@N08" ["images"]=> array(4) { [0]=> array(5) { [0]=> string(17) "Eastern_Spinebill" [1]=> string(37) "https://farm.static.flickr.com//_.jpg" [2]=> NULL [3]=> string(27) "https://flickr.com/photos//" [4]=> string(26) "https://flickr.com/people/" } [1]=> array(5) { [0]=> string(17) "Common_Bronzewing" [1]=> string(37) "https://farm.static.flickr.com//_.jpg" [2]=> NULL [3]=> string(27) "https://flickr.com/photos//" [4]=> string(26) "https://flickr.com/people/" } [2]=> array(5) { [0]=> string(12) "Little_Raven" [1]=> string(37) "https://farm.static.flickr.com//_.jpg" [2]=> NULL [3]=> string(27) "https://flickr.com/photos//" [4]=> string(26) "https://flickr.com/people/" } [3]=> array(5) { [0]=> string(5) "Galah" [1]=> string(37) "https://farm.static.flickr.com//_.jpg" [2]=> NULL [3]=> string(27) "https://flickr.com/photos//" [4]=> string(26) "https://flickr.com/people/" } } }

Notice: Undefined index: only_name in /home/pi/BirdNET-Pi/scripts/overview.php on line 62
2: 0

**Reply by ehpersonal38** [reply]
I found the culprit! A few weeks ago I went and appended "bird" to the search term that the Flickr search used. So "American Crow" would get turned into "American Crow bird" when it went to look for a matching image, to make results more relevant. This had the unfortunate result of making "user specific" searches return nothing. So I think it's fixed now, you can go ahead and update your system and report back. You'll need to clear your cookies for it to take effect (or wait 2 hours for them to expire).

**Reply by grunta56** [reply]
Yes!  Thank you, that has solved the problem. The images are back
Thanks and best regards, Grant

**Comment by krummrey** [comment]
I have made images for the top 25+ birds that seem to be around. How do I get them to load from flickr? Or should I place them on my Raspberry?
https://www.flickr.com/photos/krummrey/albums/72177720301465732
![Dendrocopos major-Buntspecht](https://user-images.githubusercontent.com/4517632/185810125-6162653b-8b2f-48d6-b7ee-817b7a424138.jpg)

**Reply by ehpersonal38** [reply]
It's going to be this option in Tools -> Settings
![chrome_q5yZY6EqpE](https://user-images.githubusercontent.com/103586016/185811415-6e8faba1-31c1-4f96-a727-bafd07a6a242.png)

Very pretty illustrations by the way. Did you paint/draw them?

**Reply by krummrey** [reply]
They are digital illustrations made with photoshop. I wanted to have a more uniform set of images. Still working on them, like having them all face right...
I entered the email from my flickr account but none of the images show up.

**Reply by krummrey** [reply]
![Bildschirmfoto 2022-08-23 um 11 28 22](https://user-images.githubusercontent.com/4517632/186123888-2e5d0b81-b115-427b-b48f-7285c3df5027.png)
Works like a charm now! BirdNetPi seems to look for the english name, not the latin one.

**Reply by ehpersonal38** [reply]
That's great! Oops yeah forgot to mention that
